demos/device_defender_for_aws/metrics_collector/lwip/metrics_collector.c

```diff
@@ -0,0 +1,209 @@
+/* Standard includes. */
+#include <stdint.h>
+/* Lwip includes. */
+#include "lwip/arch.h"
+#include "lwip/stats.h"
+#include "lwip/tcpip.h"         /* #define LOCK_TCPIP_CORE()     sys_mutex_lock(&lock_tcpip_core) */
+#include "lwip/ip_addr.h"       /*ip_addr_t, ipaddr_ntoa, ip_addr_copy */
+#include "lwip/tcp.h"           /* struct tcp_pcb */
+#include "lwip/priv/tcp_priv.h" /*tcp_listen_pcbs_t */
+/* Platform layer includes. */
+#include "platform/iot_threads.h"
+/* Demo logging include. */
+#include "iot_demo_logging.h"
+/* Interface includes. */
+#include "metrics_collector.h"
+
+#if ( FREERTOS_LWIP_METRICS_ENABLE == 1 )
+
+    extern struct tcp_pcb * tcp_active_pcbs;        /* List of all TCP PCBs that are in a state in which they accept or send data. */
+    extern union tcp_listen_pcbs_t tcp_listen_pcbs; /* List of all TCP PCBs in LISTEN state */
+
+/**
+ * @brief Get a list of the listening TCP ports.
+ *
+ * This function finds the listening TCP ports by traversing LWIP TCP PCBs list that
+ * in listen state. The header of the list is "tcp_listen_pcbs.listen_pcbs struture".
+ * It can be called with @p pOutTcpPortsArray NULL to get the number of the open TCP ports.
```

Comment:
```suggestion
 * This function can be called with @p pOutTcpPortsArray NULL to get the number of the open TCP ports.
```
sorry saying "it" sounds like you are talking about the structure from the last sentence.